libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
even when auto_reload is set to true, this will cause daemon-reload to be invoked everytime an unit is updated. should we instead use a flag and invoke daemon-reload only once, using a handler etc?